pkg/jaeger/query/trace_query.go

```diff
@@ -192,6 +200,54 @@ func (b *Builder) buildTraceIDSubquery(q *spanstore.TraceQueryParameters) (strin
 		operation_clauses = append(operation_clauses, qual)
 	}
 
+	if len(q.Tags) > 0 {
```

Comment:
I also wonder, this function is getting long. Should we have a different function for tag work? The fund name `buildTraceIDSubquery` seems verbose with tags block so long